include/libvfio-user.h

```diff
@@ -316,24 +313,33 @@ vfu_setup_device_reset_cb(vfu_ctx_t *vfu_ctx, vfu_reset_cb_t *reset);
  * Function that is called when the guest maps a DMA region. Optional.
  *
  * @vfu_ctx: the libvfio-user context
- * @iova: iova address
- * @len: length
+ * @iova: I/O virtual address and length. This is the guest physical address
+ *  range that the guest registers for DMA. This address range refers to the
+ *  guest's memory and is invalid in the user's address space. This memory is
+ *  memory mapped in the user's address space at @vaddr.
+ * @vaddr: virtual address and length if the region is mappable, which can be
+ *  different from IOVA due to alignment. If the region is not mappable then
+ *  @vaddr is NULL.
```

Comment:
Actually, this is better done under #390, and this could link to it

Reply:
the "due to alignment" bit isn't true, though: I think it should say something like:

vaddr: NULL if the region is not mapped. Otherwise, this is the virtual address and length by which the region is accessible in this process. Both values may have been aligned to a (large) page boundary. For example, an IOVA region [3MB, 6MB), once aligned, may have @vaddr range of [2MB, 6MB), and thus to access the start of the region, an offset of 1MB must be applied.